File: src/xor.go
package main

func XorEncryptDecrypt(buf []byte, key byte) []byte {
    var outBuf []byte

    for _, b := range buf {
        outBuf = append(outBuf, byte(b) ^ key)
        key = (key + 1) & 0xFF
    }

    return outBuf
}

File: decryptor/xor.c
#include <stdint.h>

void XorEncryptDecrypt(uint8_t* buf, uint8_t key, size_t bufLen) {
    for (int i = 0; i < bufLen; i++) {
        buf[i] = buf[i] ^ key;
        key = (key + 1) & 0xFF;
    }
}
